vavr/src/main/java/io/vavr/collection/Traversable.java

```diff
@@ -791,19 +803,23 @@ default T last() {
 
     /**
      * Calculates the maximum of this elements according to their natural order.
-     *
+     * <p>
+     * Examples:
+     * <pre>
+     * <code>
+     * List.empty().max()            // = None
+     * List.of(1, 2, 3).max()        // = Some(3)
+     * List.of("a", "b", "c").max()  // = Some("c")
+     * List.of(1, "a").max()         // throws
```

Comment:
Technically, this case shouldn't actually compile.

Reply:
It does...
<img width="1086" alt="screen shot 2017-09-14 at 09 10 03" src="https://user-images.githubusercontent.com/743833/30416510-bea146dc-992c-11e7-9321-0eccf6adb251.png">

Reply:
Ok, that's puzzling then - why choose serializable and comparable? That will take us off topic